x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/operator/Driver.java

```diff
@@ -181,15 +181,20 @@ private SubscribableListener<Void> runSingleLoopIteration() {
 
             if (op.isFinished() == false && nextOp.needsInput()) {
                 Page page = op.getOutput();
-                if (page == null) {
-                    // No result, just move to the next iteration
-                } else if (page.getPositionCount() == 0) {
-                    // Empty result, release any memory it holds immediately and move to the next iteration
-                    page.releaseBlocks();
-                } else {
-                    // Non-empty result from the previous operation, move it to the next operation
-                    nextOp.addInput(page);
-                    movedPage = true;
+                try {
+                    if (page == null) {
+                        // No result, just move to the next iteration
+                    } else if (page.getPositionCount() == 0) {
+                        // Empty result, release any memory it holds immediately and move to the next iteration
+                        page.releaseBlocks();
+                    } else {
+                        // Non-empty result from the previous operation, move it to the next operation
+                        nextOp.addInput(page);
+                        movedPage = true;
+                    }
+                } catch (Throwable t) {
+                    releasePageBlocksWhileHandlingException(page);
```

Comment:
I'd prefer the `success = false, success = true`  thingy. Maybe. I'm not fully sure. But it feels like a thing we're more used to seeing.

Reply:
I reverted the changes to Driver, they are not strictly needed for this change. There is an issue already tracking re-enabling the test that this change was trying to address.